### ISSUE TITLE
Remove bootstrap step

### ISSUE THREAD

**Issue by cristiklein** [elastisys/compliantkubernetes-apps#270]
**What should be investigated.**

The bootstrap step was used as a workaround for some limitation in Helm 2, specifically dealing with creation of namespaces and CRDs. However, Helm 3 handles these a lot better. Can we eliminate the bootstrap step?

**What artifacts should this produce.**

* A PR which eliminates the bootstrap step.
* Alternatively, reasons why it cannot be eliminated.

**Comment by cristiklein** [comment]
We'll wait for #242 to better understand how to move forward.

**Comment by tordsson** [comment]
Easier solvable once done with  #402 ?

**Comment by tordsson** [comment]
This is less of a problem now, as CRDs are installed with Helm.

**Comment by tordsson** [comment]
Remaining resources handled by bootstrap: storage classes + namespaces

**Comment by Xartos** [comment]
This should be possible now since we dont install any storage classes anymore (its done in the kubespray step). Namespaces can be moved to the main helmfile but be places as the first step

**Comment by cristiklein** [comment]
TL;DR: I think we make as much progress as we can with removing the bootstrapping step and should close this issue.

I just checked and I don't think it is a good idea to move namespaces to the main helmfile. The reason is that for namespaces we do `helmfile template | kubectl apply -f -` to avoid Helm taking ownership of the namespaces. In essence, we use Helmfile and Helm as templating engines but not as release management engines.

Letting Helm take ownership of namespaces would have two rather undesirable consequences:

1. We cannot set labels and annotations on post-Kubespray namespaces like `kube-system` and `kube-public`, unless we implemented a hacky bootstrap step which sets the following annotations: `app.kubernetes.io/managed-by: Helm`, `meta.helm.sh/release-name: ck8s-namespaces`, `meta.helm.sh/release-namespace: kube-system`.
2. Removing the `ck8s-namespaces` Release would essentially nuke the cluster.

No 1 is hacky and no 2 is scary, so the current solution -- i.e., keeping the bootstrap step only for namespaces, which we don't want to be owned by Helm -- feels superior.


Here is some evidence for future self:

```console
cklein@minitruee:~/elastisys/ck-exoscale-config-ng/compliantkubernetes-apps$ git status
HEAD detached at v0.20.1
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   bootstrap/bootstrap.sh

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	helmfile/00-namespaces.yaml
	helmfile/charts/namespaces/
	helmfile/values/namespaces-sc.yaml.gotmpl
	helmfile/values/namespaces-wc.yaml.gotmpl

no changes added to commit (use "git add" and/or "git commit -a")
cklein@minitruee:~/elastisys/ck-exoscale-config-ng/compliantkubernetes-apps$ git diff
diff --git a/bootstrap/bootstrap.sh b/bootstrap/bootstrap.sh
index d74fadb..7aa88d5 100755
--- a/bootstrap/bootstrap.sh
+++ b/bootstrap/bootstrap.sh
@@ -8,4 +8,4 @@ here="$(dirname "$(readlink -f "$0")")"
 
 environment="${1}"
 
-"${here}/namespaces/bootstrap.sh" "${environment}"
+#"${here}/namespaces/bootstrap.sh" "${environment}"
cklein@minitruee:~/elastisys/ck-exoscale-config-ng/compliantkubernetes-apps$ ./bin/ck8s apply wc
[ck8s] Validating wc config
[ck8s] Bootstrapping workload cluster
[ck8s] Validating wc config
[ck8s] Initializing applications
[ck8s] Applying applications in workload cluster
Creating OpenSearch and Fluentd secrets
secret/opensearch configured
secret/opensearch configured
fluentd-extra-config configmap already in place. Ignoring.
fluentd-extra-plugins configmap already in place. Ignoring.
Installing helm charts
Decrypting secret /home/cklein/elastisys/ck-exoscale-config-ng/secrets.yaml
Decrypting /home/cklein/elastisys/ck-exoscale-config-ng/secrets.yaml

Building dependency release=ck8s-namespaces, chart=charts/namespaces
Comparing release=ck8s-namespaces, chart=charts/namespaces
in ./00-namespaces.yaml: command "/home/cklein/bin/helm" exited with non-zero status:

PATH:
  /home/cklein/bin/helm

ARGS:
  0: helm (4 bytes)
  1: diff (4 bytes)
  2: upgrade (7 bytes)
  3: --reset-values (14 bytes)
  4: --allow-unreleased (18 bytes)
  5: ck8s-namespaces (15 bytes)
  6: charts/namespaces (17 bytes)
  7: --version (9 bytes)
  8: 0.1.1 (5 bytes)
  9: --namespace (11 bytes)
  10: kube-system (11 bytes)
  11: --values (8 bytes)
  12: /tmp/helmfile137953485/kube-system-ck8s-namespaces-values-6fd8b57676 (68 bytes)
  13: --detailed-exitcode (19 bytes)

ERROR:
  exit status 1

EXIT STATUS
  1

STDERR:
  Error: Failed to render chart: exit status 1: Error: rendered manifests contain a resource that already exists. Unable to continue with install: Namespace "cert-manager" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "ck8s-namespaces"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "kube-system"
  Error: plugin "diff" exited with error

COMBINED OUTPUT:
  ********************
  	Release was not present in Helm.  Diff will show entire contents as new.
  ********************
  Error: Failed to render chart: exit status 1: Error: rendered manifests contain a resource that already exists. Unable to continue with install: Namespace "cert-manager" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "ck8s-namespaces"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "kube-system"
  Error: plugin "diff" exited with error
exit status 1
```
